sys/net/application_layer/dhcpv6/client.c

```diff
@@ -119,7 +132,10 @@ static void *_thread(void *args)
     event_queue_t event_queue;
     event_queue_init(&event_queue);
     dhcpv6_client_init(&event_queue, SOCK_ADDR_ANY_NETIF);
-    /* TODO: add configuration for IA_NA here, when implemented */
```

Comment:
Should the auto-initialization also cover IA_PD?

Reply:
Mh no, the configuration for IA_PD must come from the app, as things like the network interface the delegated prefix are to be assigned to as well as the length of the prefix are configurable. For IA_NA the things are more clear, as the non-temporary address is usually assigned to the interface the DHCPv6 client is running on. Since it is currently `SOCK_ADDR_ANY_NETIF` we have to figure out what this could mean, but I believe just picking the first interface as done below is definitely wrong ;-). As such, maybe having a `void dhcpv6_client_req_ia_na(unsigned netif)` function as well could make sense.

Reply:
Alternatively, see https://github.com/RIOT-OS/RIOT/pull/16228#discussion_r673805245